curriculum/challenges/english/14-responsive-web-design-22/learn-more-about-css-pseudo-selectors-by-building-a-balance-sheet/step-031.md

```diff
@@ -1,26 +1,34 @@
 ---
-id: 6193f7904dc7ee554d12571f
+id: 61fdac1e31692f9a9ad97295
 title: Step 31
 challengeType: 0
 dashedName: step-31
 ---
 
 # --description--
 
-The `AcmeWidgetCorp` text should be a bit smaller. Create a `#header h2` selector to target only the `h2` elements within your `#header` element, and give it a `font-size` property set to `1.3em`.
+The CSS `clip` property is used to define the visible portions of an element. Set the span[class~="sr-only"] ` selector to have a `clip` property of `rect(1px, 1px, 1px, 1px)`.
+
+The `clip-path` property determines the shape the `clip` property should take. Use both the `clip-path` and `-webkit-clip-path` selectors to set the value to `inset(50%)`, forming the clip path into a rectangle within the element.
 
 # --hints--
 
-You should have a new `header h2` selector.
+Your span[class~="sr-only"] ` selector should have a `clip` property set to `rect(1px, 1px, 1px, 1px)`.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle(span[class~="sr-only"] ')?.getPropertyValue('clip') === 'rect(1px, 1px, 1px, 1px)');
+```
+
+Your span[class~="sr-only"] ` selector should have a `clip-path` property set to `inset(50%)`.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('#header h2'));
+assert(new __helpers.CSSHelp(document).getStyle(span[class~="sr-only"] ')?.getPropertyValue('clip-path') === 'inset(50%)');
 ```
 
-Your `header h2` selector should have a `font-size` property set to `1.3em`.
+Your span[class~="sr-only"] ` selector should have a `-webkit-clip-path` property set to `inset(50%)`.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('#header h2')?.fontSize === '1.3em');
+assert(new __helpers.CSSHelp(document).getStyle(span[class~="sr-only"] ')?.getPropertyValue('-webkit-clip-path') === 'inset(50%)');
```

Comment:
```suggestion
assert(new __helpers.CSSHelp(document).getStyle('span[class~="sr-only"]')?.getPropertyValue('-webkit-clip-path') === 'inset(50%)');
```
However this still doesn't work on Firefox, I wasn't able to find alternative working property for it. `.webkitClipPath` also doesn't work on ff.

Reply:
Any thoughts on how to fix this? 🤔

Reply:
Regex might be the only resort. What adds to strangeness, is that on chrome following css will pass
```css
span[class~="sr-only"] {
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  -webkit-clip-path: inset(50%);
}
```